# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/atk/AccessibleWrap.cpp
# Commit: 8d31c0604efe
# Full Hash: 8d31c0604efef2015995df32766d38dd4dd45ca2
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-08-04 04:34:13
# Regressor Bug: 1781536
# File Overlap Count: 1
# Description:
#   Bug 1781536 part 1: Support base Accessible in Relation. r=morgan
#   
#   Previously, Relation::Next returned a LocalAccessible.
#   Now, it returns an Accessible so that it can support RemoteAccessible as well.
#   Relation::LocalNext has been added as a convenience to return a LocalAccessible when the caller is certain that only LocalAccessible is relevant.
# ==============================================================================

diff -r af7f9589f48c -r 8d31c0604efe accessible/atk/AccessibleWrap.cpp
--- a/accessible/atk/AccessibleWrap.cpp	Thu Aug 04 02:47:34 2022 +0300
+++ b/accessible/atk/AccessibleWrap.cpp	Wed Aug 03 23:58:50 2022 +0000
@@ -817,9 +817,9 @@
 
   Relation rel(aAcc->RelationByType(aType));
   nsTArray<AtkObject*> targets;
-  LocalAccessible* tempAcc = nullptr;
+  Accessible* tempAcc = nullptr;
   while ((tempAcc = rel.Next())) {
-    targets.AppendElement(AccessibleWrap::GetAtkObject(tempAcc));
+    targets.AppendElement(GetWrapperFor(tempAcc));
   }
 
   if (aType == RelationType::EMBEDS && aAcc->IsRoot()) {